core/templates/base-components/oppia-footer.component.ts

```diff
@@ -53,6 +63,40 @@ export class OppiaFooterComponent {
       return 'https://medium.com/oppia-org';
     }
   }
+
+  validateEmailAddress(): boolean {
+    let regex = new RegExp(AppConstants.EMAIL_REGEX);
+    return regex.test(String(this.emailAddress));
+  }
+
+  subscribeToMailingList(): void {
```

Comment:
I have some concerns about this:

- We removed this functionality from the donate page. So why isn't it deleted in this PR?
- How did you test that this works? There should be frontend tests, but I don't see a spec file.

Reply:
I am sorry :(, but I am not able to quite understand which functionality you are talking about `subscribeToMailingList` or `validateEmailAddress`. Could you please elaborate a bit more and the test file?

Reply:
Yup. You removed the newsletter section from the HTML in the donate page. So shouldn't there be some removal of newsletter-signup TypeScript code (that should now be dead code since it's no longer used)?

Reply:
Thank You. Sorry, I am not able to find any unused code in `donate-page.component.ts` and `donate-page.component.spec.ts`. Could you please point me to any unused code? I believe I removed the sign-up ts logic from both files.